### ISSUE TITLE
question. bind to array works ?

### ISSUE THREAD

**Issue by cesarchefinho** [levitali/CompiledBindings#21]
Does this works with compiledbindings ?

// ViewModel

public string ErrorOf [string index]
{
    // get and set accessors
}

// XAML

< textblock Text={x:Bind ViewModel.ErrorOf ["Username"]}

if yes, please considere to add binding to arrays in readme, if no please considere open an issue

UWP Bind supports bind to array property

**Comment by levitali** [comment]
Yes, you can access elements of an array or of the Item property.

The c# and xaml code in your example is not correct. You can't have named Indexer property in c#. Instead you can have something like this:

 ```c#
public class ErrorsCollection
{
	public string this[string index]
	{
		get => // some logic
	}
}

class ViewModel
{
	public ErrorsCollection ErrorOf { get; }
}
 ```

 ```xaml
<TextBlock Text="{x:Bind ViewModel.ErrorOf['Username']}"/>
 ```

**Comment by cesarchefinho** [comment]
and what about IndexerName attribute ?

UWP and WINUI recognize it in x:Bind

class ViewModel 
{
....
    [IndexerName("ErrorOf")]
    public string this[string columnName] =>
        string.Join(Environment.NewLine,
                    (from ValidationResult e 
                       in GetErrors(columnName) 
                   select e.ErrorMessage));

    public void OnErrorsChanged(object sender, DataErrorsChangedEventArgs e)
    {
        if (string.IsNullOrWhiteSpace(e.PropertyName))
        {
            OnPropertyChanged("ErrofOf['']");
        }
        else
        {
            OnPropertyChanged($"ErrorOf['{e.PropertyName}']");
        }
    }

**Comment by cesarchefinho** [comment]
Does compiled bindings recognize [IndexerName] attribute ?

**Comment by levitali** [comment]
IndexerNameAttribute currently is not supported. I'll add it in the next version. Now the property with "Item" name is searched to find the indexer property.

IndexerNameAttribute is actually meant to be able to consume c# libraries in other programming languages. If your code is completely in c#, setting the attribute brings you nothing. You can’t use the name neither in Bindings nor in UWP/WinUI  x:Binds.

In your OnErrorsChanged you can just raise PropertyChanged event with any string. The generated code doesn’t check property names, just updates Binding on any event.

In UWP for this XAML:
``` xaml
<Grid>
  <TextBlock Text="{x:Bind _viewModel[3], Mode=OneWay}"/>
  <TextBlock Text="{x:Bind _viewModel[2], Mode=OneWay}"/>
</Grid>
```
this c# code is generated:´

``` c#

public void CollectionChanged__viewModel(object sender, global::System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
{
     MainPage_obj1_Bindings bindings = TryGetBindingObject();
     if (bindings != null)
     {
          global::App7.MainViewModel obj = sender as global::App7.MainViewModel;
           if (obj != null)
          {
                bindings.Update__viewModel_I3(obj[3], DATA_CHANGED);
                bindings.Update__viewModel_I2(obj[2], DATA_CHANGED);
          }
     }
}
```

In my CompiledBindings notifications for element access expressions are currently not supported. I will also implement it in the next version.
